### PR TITLE
improvement: add casing version and allow for "preserved" casing words

### PR DESCRIPTION
**Pull request by armandobelardo**

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts [hunk 1]
```diff
@@ -37,8 +39,52 @@ export function constructCasingsGenerator({
...
             let snakeCaseName = snakeCase(name);
-            const camelCaseWords = words(camelCaseName);
-            if (smartCasing) {
+
+            // generatorsYml.CasingVersion.V0 should effectively be treated as smartCasing
+            if (
+                smartCasing ||
+                casingVersion === generatorsYml.CasingVersion.V0 ||
+                casingVersion === generatorsYml.CasingVersion.V1
+            ) {
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:48]: i would have thought that `v0` would not default to smart casing? 

Some context on the PR about what is `v0` and what is `v1` would be helpful

File: packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts [hunk 2]
```diff
@@ -18,11 +18,13 @@ const CAPITALIZE_INITIALISM: generatorsYml.GenerationLanguage[] = ["go", "ruby"]
...
     generationLanguage,
     keywords,
-    smartCasing
+    smartCasing,
+    casingVersion
 }: {
     generationLanguage: generatorsYml.GenerationLanguage | undefined;
     keywords: string[] | undefined;
     smartCasing: boolean;
+    casingVersion: generatorsYml.CasingVersion | undefined;
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:27]: do you think we can delete smartCasing altogether and just rely on `casingVersion` now  here? 

i would have expected the code to read something like: 
```ts
switch (version) {
  case "v0": 
    // do a vanilla case
  case "v1": 
    // handle initialisms and numbers
  default: 
    assertNever(version)
}
```

Comment (armandobelardo) [reply packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:27]: So you want to have a yaml migration to move smartCasing to v0?

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:27]: I think internally in the codebase we can have `casingVersion` (in `ir-generator` for example), but externally we can listen to version first and then fall back to smart-casing -> v1

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:27]: so no yaml migration needed

**Pull request closed without merging**